web/embedding.py

```diff
@@ -104,21 +104,28 @@ def transform_words(self, f, inplace=False):
                     # Overwrites with last occurence
                     self.vectors[id_map_to_new[fw]] = self.vectors[id]
 
-
             words = sorted(id_map.keys(), key=lambda x: id_map[x])
             self.vectors = self.vectors[0:len(id_map)]
             self.vocabulary = self.vocabulary.__class__(words)
+
+            self.vocabulary, self.vectors = Embedding._deduplicate(self.vocabulary, self.vectors)
+
             logger.info("Tranformed {} into {} words".format(word_count, len(words)))
             return self
         else:
             for id, w in enumerate(self.vocabulary.words):
                 if len(f(w)) and (f(w) not in id_map or f(w) == w):
                     id_map[f(w)] = id
+
             words = sorted(id_map.keys(), key=lambda x: id_map[x])
             vectors = self.vectors[[id_map[w] for w in words]]
+
+            words, vectors = Embedding._deduplicate(words, vectors)
```

Comment:
W fetcherach. W konstruktorze niech będzie wyjątek (jak juz jest to sorry, my bad)